pyblish/plugin.py

```diff
@@ -666,6 +666,14 @@ def __call__(self, key=None, default=None):
 
         return self.get(key, default)
 
+    def __setitem__(self, k, v):
+        # Validate "publish" data member to always be boolean
+        if k == "publish":
+            msg = "\"publish\" data member has to be boolean."
+            assert isinstance(v, bool), msg
```

Comment:
Would a `TypeError` be more intuitive here?

Reply:
Good point! Done.